### PR TITLE
M2P-211 Add support for module MageWorld RewardPoints

### PR DESCRIPTION
**Pull request by dumega**
# Description
Add support for module MageWorld RewardPoints

Fixes: https://boltpay.atlassian.net/browse/M2P-211

#changelog Add support for module MageWorld RewardPoints

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [x] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: Block/Js.php [hunk 1]
```diff
@@ -205,8 +214,8 @@ public function getJavascriptSuccess()
      * @return string
      */
     public function getAdditionalJavascript()
-    {
-        return $this->configHelper->getAdditionalJS();
+    {
```
Comment (vitaliyreznikov) [inline Block/Js.php:217]: nit: remove blanks at the end of string

**Review by vitaliyreznikov — COMMENTED** [review]
Looks good! Have you tested it with update_order call? Are we able to delete this reward points?

File: ThirdPartyModules/MageWorld/RewardPoints.php [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    public function beforePrepareQuote($mwRewardPointsHelperData,
+                                     $mwRewardPointsModelCustomer,
+                                     $quote)
+    {
+        $this->mwRewardPointsHelperData = $mwRewardPointsHelperData;
+        $this->mwRewardPointsModelCustomer = $mwRewardPointsModelCustomer;
+        
+        try {
+            if ($quote->getMwRewardpoint()) {
+                $currencyCode = $quote->getQuoteCurrencyCode();
```
Comment (johnmccombs1) [inline ThirdPartyModules/MageWorld/RewardPoints.php:152]: @dumega I'm not seeing $currencyCode used anywhere in this class. Should we remove it?

File: ThirdPartyModules/MageWorld/RewardPoints.php [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+namespace Bolt\Boltpay\ThirdPartyModules\MageWorld;
+
+use Bolt\Boltpay\Helper\Bugsnag;
+use Bolt\Boltpay\Helper\Discount;
+use Bolt\Boltpay\Helper\Shared\CurrencyUtils;
+
+class RewardPoints
+{    
+    /**
+     * @var DiscountHelper
```
Comment (johnmccombs1) [inline ThirdPartyModules/MageWorld/RewardPoints.php:27]: @dumega It should be Discount, right?

File: Test/Unit/Block/JsTest.php [hunk 4]
```diff
@@ -249,7 +256,8 @@ public function __construct_always_setsInternalProperties()
             $this->checkoutSessionMock,
             $this->cartHelperMock,
             $this->bugsnagHelperMock,
-            $this->deciderMock
+            $this->deciderMock,
+            $this->eventsForThirdPartyModules
         );
         static::assertAttributeEquals($this->configHelper, 'configHelper', $instance);
```
Comment (johnmccombs1) [inline Test/Unit/Block/JsTest.php:262]: **We should test the $eventsForThirdPartyModules argument. Something like this:**
 
static::assertAttributeEquals($this-> eventsForThirdPartyModules, 'eventsForThirdPartyModules', $instance);

**Review by johnmccombs1 — CHANGES_REQUESTED** [review]

File: Block/Js.php (hunk 1 above)
Comment (dumega) [reply Block/Js.php]: fixed

File: ThirdPartyModules/MageWorld/RewardPoints.php (hunk 2 above)
Comment (dumega) [reply ThirdPartyModules/MageWorld/RewardPoints.php]: removed

File: ThirdPartyModules/MageWorld/RewardPoints.php (hunk 3 above)
Comment (dumega) [reply ThirdPartyModules/MageWorld/RewardPoints.php]: fixed

File: Test/Unit/Block/JsTest.php (hunk 4 above)
Comment (dumega) [reply Test/Unit/Block/JsTest.php:262]: added

**Comment by dumega** [comment]
> Looks good! Have you tested it with update_order call? Are we able to delete this reward points?

@vitaliyreznikov for now v2 update_cart does not support to remove reward points.

**Review by danac-gs — DISMISSED** [review]

**Review by johnmccombs1 — APPROVED** [review]

**Comment by vitaliyreznikov** [comment]
@dumega We can enable removing reward points, so I created the task to implement it for this plugin
https://boltpay.atlassian.net/browse/M2P-253

**Review by vitaliyreznikov — APPROVED** [review]
Please fix the conflict.

**Review by matt-thomason — APPROVED** [review]

**Pull request merged**